### ISSUE TITLE
Regarding the results of training on a single GPU

### ISSUE THREAD

**Issue by SYDDX** [xupei0610/KDMA#11]
Hello, your work has been very inspiring to me. However, I encountered a problem during training. Since I can only use a personal computer for training and cannot perform distributed training, I modified the training command to: python train_policy.py --rank 0 --workers 1 --device cuda:0 --expert expert --log_dir policy, where expert is the directory of the expert policy model and policy is the directory for the policy model trained by RL. I successfully started the training, but the results are very poor. The terminal displays the following:
![QQ_1725278778894](https://github.com/user-attachments/assets/964a1c13-2553-44d6-b4f7-e49d19650be5)
Why does the collision probability increase and the success rate decrease the longer I train? This seems completely opposite to the expected results! Additionally, I have encountered a minor issue:
![QQ_1725278973807](https://github.com/user-attachments/assets/e2ac4277-cbe0-48ec-a610-db966fa81387)
What is causing this issue? I would really appreciate it if you could help me with my question.

**Comment by xupei0610** [comment]
I would recommend performing training with distributed environments. This is crucial to increase the diversity of training samples. If your GPU does not have enough memory, you probably can try to run training on CPU.

The PytorchStreamReader error is trivial. It means the checkpoint file is modified when the evaluator thread try to read it for evaluation.

**Comment by SYDDX** [comment]
Thank you for your suggestion! I have another question. When I use the above command line (single GPU) for training, an error occurs in the **update** function of the **actor_critic.py** file:
![QQ_1725328273216](https://github.com/user-attachments/assets/0e2e466c-41b0-43b9-bd45-de5783034536)
Could this also be due to single GPU training? To try to solve this problem, I made the following modifications:
![QQ_1725329807385](https://github.com/user-attachments/assets/839be3d3-03bb-47ed-a4ab-ad458eee5ef7)
After that, the training was able to proceed, but the results were completely opposite to what I expected, as I described earlier. Is there an issue with my modifications?
